apps/core/scenes/search/PlacePicker/PlaceItem.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import {
+  Text,
+  StyleSheet,
+  Icon,
+  Touchable,
+} from '@kiwicom/universal-components';
+
+import { type Location } from '../SearchContext';
+
+type Props = {|
+  +onPress: Location => void,
+  +location: ?Location,
+|};
+
+class PlaceItem extends React.Component<Props> {
+  handlePress = () => {
+    const { location, onPress } = this.props;
+    location && onPress(location); // eslint-disable-line babel/no-unused-expressions
```

Comment:
Please don't disable eslint-rules like this, they are there for a reason. 

Also, I don't see why we should props-drill the on press handler down 3 levels when the drilled function just calls functions in the context. 

Just connect to the context in this component.

Reply:
Improved, but I need this pass down through 3 levels because the `onPress(location)` is not the same for every use cases.

Reply:
> I need this pass down through 3 levels because the `onPress(location)` is not the same for every use cases

This is the current click handler

```
const { type, setTravelFrom, setTravelTo, setModalType } = this.props;

    if (type === MODAL_TYPE.ORIGIN) {
      setTravelFrom(location);
    }
    if (type === MODAL_TYPE.DESTINATION) {
      setTravelTo(location);
    }
    setModalType(MODAL_TYPE.HIDDEN);
```

All it does is check some values in the context, then call some values in the context. There is no need for this logic to be defined here. It has nothing to do with this component. 

When I want to know what happens when I click on something, I will look for the click handler in the component where the button is defined, then I have to go back 3 levels to find out what really happens. 

Please move this function to `PlaceItem`

Reply:
OK, I will change it, but with this approach how you look on *presentational and container components* pattern?

Reply:
I think this concept is often misunderstood, and done wrong. Just look at our FE where there are `Containers` that are 1000+ lines of code.

You also might see What Dan Abramov says about it in [this](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0) old article.

## When to Introduce Containers?
> When you notice that some components don’t use the props they receive but merely forward them down

## Presentational 
• Don’t specify how the data is loaded or mutated
• Receive data and callbacks exclusively via props

## Benefits of This Approach
• You can use the same presentational component with completely different state sources


I would say our presentational components are `@kiwicom/margarita-components` and `@kiwicom/universal-components`. Of course if `PlacePicker.js` was to be reused by some other component, then I would agree it should have the onPress handler passed as a prop. So if some day in the future some other component will reuse this, we could refactor it to take it as a prop.

Reply:
Thanks for the explanation. I am still not 100% sure where is the border, but
 
> I would say our presentational components are `@kiwicom/margarita-components` and `@kiwicom/universal-components`.

make sense for me.